src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Models/CommonModels/PolicyRetentionObjects.cs

```diff
@@ -23,19 +23,28 @@ namespace Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets.Models
 {
     public class AzureRmRecoveryServicesBackupSimpleRetentionPolicy : AzureRmRecoveryServicesBackupRetentionPolicyBase
     {
-        public RetentionDuration RetentionDuration { get; set; }
+        public RetentionDurationType RetentionDurationType { get; set; }
+
+        public int RetentionCount { get; set; }
 
         public override string ToString()
         {
             return string.Format("RetentionDurationType: {0}, Retention Count : {1}",
-                        RetentionDuration.RetentionDurationType.ToString(),
-                        RetentionDuration.RetentionCount);            
+                        RetentionDurationType.ToString(),
+                        RetentionCount);            
         }
 
         public override void Validate()
         {
             base.Validate();
-            RetentionDuration.Validate();
+
+            if ((RetentionDurationType == RetentionDurationType.Days) ||
+            (RetentionDurationType == RetentionDurationType.Weeks && (RetentionCount <= 0 || RetentionCount > PolicyConstants.MaxAllowedRetentionDurationCountWeeklySql))||
```

Comment:
Not taking this one. This is a complex condition, and I have tried to keep one condition in one like, like condition related to weeks in one line, condition related to months in another.